### PR TITLE
Implement new wireguard-go

### PR DESCRIPTION
**Pull request by peske**
Registry-access code replaced as sugested. Now it uses `x/sys/windows`
Adapted to changed `wireguard-go` API.

### CODE REVIEW

File: wgengine/router/dns/manager_windows.go [hunk 1]
```diff
@@ -35,14 +36,76 @@ func newManager(mconfig ManagerConfig) managerImpl {
...
+	// appear. For some reason, registry keys tied to ephemeral interfaces
+	// can take a long while to appear after interface creation, and we
+	// can end up racing with that.
+	keyOpenTimeout = time.Minute
+
+	// REG_NOTIFY_CHANGE_NAME notifies the caller if a subkey is added or deleted.
+	REG_NOTIFY_CHANGE_NAME uint32 = 0x00000001
+)
+
+func openKeyWait(k registry.Key, path string, access uint32, timeout time.Duration) (registry.Key, error) {
```
Comment (josharian) [inline wgengine/router/dns/manager_windows.go:50]: We need to put any code copied from wireguard in its own file with the wireguard license at the top. (Perhaps later it'll get upstreamed into x/sys/windows.)

File: cmd/tailscaled/depaware.txt [hunk 2]
```diff
@@ -99,7 +94,7 @@ tailscale.com/cmd/tailscaled dependencies: (generated by github.com/tailscale/de
         tailscale.com/wgengine/tstun                                 from tailscale.com/wgengine
    W 💣 tailscale.com/wgengine/winnet                                from tailscale.com/wgengine/router
         golang.org/x/crypto/blake2b                                  from golang.org/x/crypto/nacl/box
-        golang.org/x/crypto/blake2s                                  from github.com/tailscale/wireguard-go/device+
+        golang.org/x/crypto/blake2s                                  from github.com/tailscale/wireguard-go/device
```
Comment (josharian) [inline cmd/tailscaled/depaware.txt:97]: This is also fine as-is, but FYI, if you install the latest version of depaware (as of an hour or so ago), it'll generate a smaller diff for you.

File: wgengine/router/dns/manager_windows.go (hunk 1 above)
Comment (peske) [reply wgengine/router/dns/manager_windows.go:50]: OK, will do it in a minute.

**Comment by josharian** [comment]
Once the code from wireguard is handled appropriately, license-wise, this LGTM. Let me know what I should take another look.

**Comment by josharian** [comment]
By the way, if you want to avoid having to make new PRs, you can rebase down to a clean set of changes and then do a force push of the branch.

File: cmd/tailscaled/depaware.txt (hunk 2 above)
Comment (peske) [reply cmd/tailscaled/depaware.txt:97]: Didn't know. Will update. Thanks!

**Comment by peske** [comment]
> 
> 
> Once the code from wireguard is handled appropriately, license-wise, this LGTM. Let me know what I should take another look.

Well, I don't see anything tricky that requires special attention...

**Comment by josharian** [comment]
> Well, I don't see anything tricky that requires special attention...

Hahah. I meant to type "when I should take another look", not "what". :P

**Comment by peske** [comment]
> 
> 
> By the way, if you want to avoid having to make new PRs, you can rebase down to a clean set of changes and then do a force push of the branch.

If it is my choice - I would go with PRs...

**Comment by josharian** [comment]
> If it is my choice - I would go with PRs...

I mean do a force push of the `peske/new-wireguard-go` branch.

**Comment by bradfitz** [comment]
In the future don't merge a commit series where each commit doesn't build and pass on its own. It makes bisect painful. We prefer squash and merge so there's just one rebased commit and linear history if it's all one logical change.

**Pull request merged**